### ISSUE TITLE
If an install succeeds but an instantiate fails, you are left with a package on the peer and cannot deploy again

### ISSUE THREAD

**Issue by davidkel** [hyperledger/composer#685]
```
Error: Error installing chaincode code carauction-network3:0.5.10-20170410155533(chaincode /var/hyperledger/production/chaincodes/carauction-network3.0.5.10-20170410155533 exists)
Command failed
```

So long as that package is ok, then we could ignore the error on the install part that the package already exists and just move on to the instantiateProposal.

If the package is not ok, currently there is no way to delete it (but that may be coming) however if it wasn't ok then it should be a newer version of the business network. ~~The Current signature only includes the level of composer runtime and the name of the business network but should also include the version of the business network as well.~~ (after discovering the network upgrade facility the previous thought isn't applicable so discarding)

**Comment by sstone1** [comment]
> The Current signature only includes the level of composer runtime and the name of the business network but should also include the version of the business network as well.

The business network definition (and it's version) can be updated using chaincode calls without upgrading the chaincode itself. Are you suggesting that we should perform a chaincode upgrade (needed if the version is Composer version + business network version) each time we update the business network definition? That seems a bit overkill!

We should be installing the Composer chaincode once per Composer version, with a generic ID - e.g. `hyperledger-composer@0.5.11`. The only time we should upgrade the chaincode is when a new Composer version comes along and the user wants to pick it up.

Unfortunately that's not possible yet, but I discussed with the Fabric team last week and raised a JIRA issue: https://jira.hyperledger.org/browse/FAB-3016

In the mean time I think it would be safe to ignore the chaincode exists error and carry on with the instantiate.

**Comment by davidkel** [comment]
Is it possible to run multiple business networks within a single chaincode container or is that something we are considering in the future ? If we used a generic id for the chaincode then would someone be able to deploy multiple business networks and have them run in different containers ? on the same peers ?

**Comment by sstone1** [comment]
It is not currently possible, and I would not like to introduce support for multiple business networks running within a single chaincode. Separation by using multiple chain codes gives us true isolation (different world state) between multiple business networks.

**Comment by caroline-church** [comment]
@davidkel @sstone1 What actions do we need to take to fix this?

**Comment by davidkel** [comment]
this has now been fixed by #904
